lib/charms/postgresql_k8s/v0/postgresql.py

```diff
@@ -428,8 +431,10 @@ def _generate_database_privileges_statements(
                     SQL("GRANT ALL PRIVILEGES ON ALL FUNCTIONS IN SCHEMA {} TO {};").format(
                         schema, Identifier(user)
                     ),
-                    SQL("GRANT USAGE ON SCHEMA {} TO {};").format(schema, Identifier(user)),
-                    SQL("GRANT CREATE ON SCHEMA {} TO {};").format(schema, Identifier(user)),
+                    SQL("GRANT USAGE, CREATE ON SCHEMA {} TO {};").format(
+                        schema, Identifier(user)
+                    ),
+                    SQL("GRANT USAGE, CREATE ON SCHEMA {} TO admin;").format(schema),
```

Comment:
I checked the [PostgreSQL 15 release notes](https://www.postgresql.org/docs/release/15.0/) and changed my mind about the approach here. Now, I think that it's important that we don't grant those permissions to users on the public schema to avoid CVE-2018-1058.